pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,619 @@
+        if i <= len(args) - 3:
+            pstr += ','
+        else:
+            pstr += ')'
+    return pstr
+
+
+def handle_functionID_node(node, *args):
+    # seems to just be a placeholder empty wrapper object
+    return ''
+
+
+def handle_indexTemplate_node(node, *args):
+    return '__INDEX_PLACEHOLDER_8675309_GROUP_%s_%s__' % (node._group, node._set)
+
+
+def handle_numericGIE_node(node, *args):
+    pstr = ''
+    pstr += args[0] + '_{'
+    for i in range(1, len(args)):
+        pstr += args[i]
+        if i <= len(args) - 2:
+            pstr += ','
+        else:
+            pstr += '}'
+    return pstr
+
+
+def handle_templateSumExpression_node(node, *args):
+    pstr = ''
+    pstr += '\\sum_{%s} %s' % (
+        '__SET_PLACEHOLDER_8675309_GROUP_%s_%s__'
+        % (node._iters[0][0]._group, str(node._iters[0][0]._set)),
+        args[0],
+    )
+    return pstr
+
+
+class _LatexVisitor(StreamBasedExpressionVisitor):
+    def __init__(self):
+        super().__init__()
+
+        self._operator_handles = {
+            ScalarVar: handle_var_node,
+            int: handle_num_node,
+            float: handle_num_node,
+            NegationExpression: handle_negation_node,
+            ProductExpression: handle_product_node,
+            DivisionExpression: handle_division_node,
+            PowExpression: handle_pow_node,
+            AbsExpression: handle_abs_node,
+            UnaryFunctionExpression: handle_unary_node,
+            Expr_ifExpression: handle_exprif_node,
+            EqualityExpression: handle_equality_node,
+            InequalityExpression: handle_inequality_node,
+            RangedExpression: handle_ranged_inequality_node,
+            _GeneralExpressionData: handle_named_expression_node,
+            ScalarExpression: handle_named_expression_node,
+            kernel.expression.expression: handle_named_expression_node,
+            kernel.expression.noclone: handle_named_expression_node,
+            _GeneralObjectiveData: handle_named_expression_node,
+            _GeneralVarData: handle_var_node,
+            ScalarObjective: handle_named_expression_node,
+            kernel.objective.objective: handle_named_expression_node,
+            ExternalFunctionExpression: handle_external_function_node,
+            _PythonCallbackFunctionID: handle_functionID_node,
+            LinearExpression: handle_sum_expression,
+            SumExpression: handle_sum_expression,
+            MonomialTermExpression: handle_monomialTermExpression_node,
+            IndexedVar: handle_var_node,
+            IndexTemplate: handle_indexTemplate_node,
+            Numeric_GetItemExpression: handle_numericGIE_node,
+            TemplateSumExpression: handle_templateSumExpression_node,
+        }
+
+    def exitNode(self, node, data):
+        return self._operator_handles[node.__class__](node, *data)
+
+
+def latex_printer(
+    pyomoElement, filename=None, useAlignEnvironment=False, splitContinuousSets=False
```

Comment:
I think at least for user-facing variables, using snake case would be nice for the sake of consistency. (And again, I'd advocate for `pyomo_component`)